lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -100,23 +154,28 @@ Ufe::ContextOps::Items UsdContextOps::getItems(
 {
     Ufe::ContextOps::Items items;
     if (itemPath.empty()) {
-        // Top-level items.  Variant sets and visibility.
-        if (fPrim.HasVariantSets()) {
-            items.emplace_back(
-                "Variant Sets", "Variant Sets", Ufe::ContextItem::kHasChildren);
-        }
-        auto attributes = Ufe::Attributes::attributes(sceneItem());
-        if (attributes) {
-            auto visibility =
-                std::dynamic_pointer_cast<Ufe::AttributeEnumString>(
-                    attributes->attribute(UsdGeomTokens->visibility));
-            if (visibility) {
-                auto current = visibility->get();
-                const std::string l = (current == UsdGeomTokens->invisible) ?
-                    std::string("Make Visible") : std::string("Make Invisible");
-                items.emplace_back("Toggle Visibility", l);
+        // Top-level items.  Variant sets and visibility. Do not add for gateway type node.
+        if (!fIsAGatewayType) {
+            if (fPrim.HasVariantSets()) {
+                items.emplace_back(
+                    "Variant Sets", "Variant Sets", Ufe::ContextItem::kHasChildren);
+            }
+            auto attributes = Ufe::Attributes::attributes(sceneItem());
+            if (attributes && attributes->hasAttribute(UsdGeomTokens->visibility)) {
```

Comment:
Added the hasAttribute() test here to fix a bug with prims that don't have a visibility attribute to toggle.

Reply:
Thanks.  For my own education, which ones don't?  The typeless defs?

Reply:
Typeless = yes, but also shaders, materials

Reply:
Ah, of course.  Can you add that in as a code comment, to avoid further questions?